package.json

```diff
@@ -3,42 +3,44 @@
   "private": true,
   "version": "3.5.7",
   "description": "🐻 Bear necessities for state management in React",
-  "main": "index.js",
-  "module": "esm/index.js",
-  "types": "index.d.ts",
+  "main": "./index.js",
+  "module": "./esm/index.mjs",
```

Comment:
Do we need this line? Any bundlers make use of it?

Reply:
Webpack and rollup use it.
pure esm version will pickup the `exports` part , so you can say it's for `backward compat`

Reply:
Oh, yeah. and those bunders don't complain with `mjs` extension, right?

Reply:
Webpack reads it as proper esm module but like in https://github.com/pmndrs/zustand/issues/454 if the source is incompatible with esm then the fix provided there will have to be added to webpack 4

And other bundlers that don't support esm or modules will not touch the `/esm` at all so it should be fine. 

Right now there's 3 ways to import 


```js

import {} from "zustand/middleware" // works on both esm and cjs based on the package.json / bundler
import {} from "zustand/esm/middleware" // force esm for certain bundlers (tried on parcel)
import {} from "zustand/esm/middleware.mjs" // shove the esm down the bundler if needed

```

And the `package.json` forces `type:module` in the later two cases so if the bundler doesn't like esm, then it's on the bundler not zustand as a package.